Search.setIndex({"alltitles": {"1. Primer Format (Regular expression based)": [[0, "primer-format-regular-expression-based"]], "16SAmpliconKit": [[4, null]], "2. Naming Requirements": [[0, "naming-requirements"]], "3. Snakemake Errors": [[0, "snakemake-errors"]], "4. QIIME 2 Errors": [[0, "qiime-2-errors"]], "5. Taxonomic Classifiers": [[0, "taxonomic-classifiers"]], "Additional Information": [[0, null]], "Directory Structure Overview": [[1, null]], "Fetching Your Own Data": [[2, null]], "Installation": [[3, null]], "Running the Pipeline": [[5, null]], "Step 1: Choose Paired-End or Single-End Data": [[2, "step-1-choose-paired-end-or-single-end-data"]], "Step 1: Clone the repository": [[3, "step-1-clone-the-repository"]], "Step 1: Navigate to the rawreads/ directory based on data type": [[5, "step-1-navigate-to-the-rawreads-directory-based-on-data-type"]], "Step 1: Navigate to the test_data Directory": [[6, "step-1-navigate-to-the-test-data-directory"]], "Step 2: Activate the snakemake environment and execute the Snakefile": [[6, "step-2-activate-the-snakemake-environment-and-execute-the-snakefile"]], "Step 2: Editing the Fetch Script: Add Your BioProject and Sample IDs": [[2, "step-2-editing-the-fetch-script-add-your-bioproject-and-sample-ids"]], "Step 2: Make the scripts executable": [[3, "step-2-make-the-scripts-executable"]], "Step 2: Update the primer sequence based on the amplicon region": [[5, "step-2-update-the-primer-sequence-based-on-the-amplicon-region"]], "Step 3: Make the Script Executable": [[2, "step-3-make-the-script-executable"]], "Step 3: Make the Scripts Executable": [[5, "step-3-make-the-scripts-executable"]], "Step 3: Set up the conda environment and install tools": [[3, "step-3-set-up-the-conda-environment-and-install-tools"]], "Step 4: Auto-Generate Snakefiles for Each BioProject": [[5, "step-4-auto-generate-snakefiles-for-each-bioproject"]], "Step 4: Run the Script to Download Raw Reads": [[2, "step-4-run-the-script-to-download-raw-reads"]], "Step 5: Execute the Snakefile in each folder": [[5, "step-5-execute-the-snakefile-in-each-folder"]], "Testing the installation on a example dataset": [[6, null]]}, "docnames": ["additional_info", "directory_str", "fetch_data", "installation", "intro", "run_pipeline", "test_pipeline"], "envversion": {"sphinx": 62, "sphinx.domains.c": 3, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 9, "sphinx.domains.index": 1, "sphinx.domains.javascript": 3, "sphinx.domains.math": 2, "sphinx.domains.python": 4, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1}, "filenames": ["additional_info.md", "directory_str.md", "fetch_data.md", "installation.md", "intro.md", "run_pipeline.md", "test_pipeline.md"], "indexentries": {}, "objects": {}, "objnames": {}, "objtypes": {}, "terms": {"": [0, 1, 2, 6], "138": 0, "16": [3, 4, 5], "16sampliconkit": [0, 1, 2, 3], "2": [1, 4], "4": 6, "515f": 5, "806r": 5, "A": 6, "AT": [0, 5], "For": 0, "If": [0, 2, 6], "It": 4, "The": [3, 4, 5, 6], "These": [0, 3], "a1_setup_env": [1, 3, 6], "a2_fetch_paired_data": [1, 2], "a2_fetch_singlend_data": [1, 2], "abov": 0, "ac": [0, 5], "access": 5, "acg": [0, 5], "across": 4, "act": [0, 5], "activ": 0, "adapt": [1, 5, 6], "add": 5, "after": [3, 5, 6], "all": 5, "amplicon": [3, 4], "an": [2, 5], "analys": 4, "analysi": [3, 5], "analyz": 4, "ananya1006": 3, "ani": [2, 4], "appli": 4, "appropri": 5, "ar": [0, 2, 5, 6], "area": 0, "arrai": [2, 5], "associ": 2, "asv": 5, "autom": [0, 4], "automat": [3, 5], "avail": 6, "b1_create_snakefil": [1, 5], "b2_snakefile_mast": [1, 5], "base": 4, "bash": [2, 3, 5], "basic": 6, "bbduk": 5, "bbtool": 3, "been": 5, "begin": 0, "below": 5, "bioproject": [0, 4], "bioprojectid_samplesno": 2, "book": 4, "both": [2, 5], "bracket": 0, "build": 5, "built": 4, "cagc": [0, 5], "can": [2, 5, 6], "cd": [3, 5, 6], "check": 6, "chmod": [2, 3, 5], "classif": [4, 5], "classifi": [1, 5], "code": 0, "com": 3, "common": 0, "commun": 4, "compar": 4, "compat": 1, "complet": [5, 6], "comput": 4, "conda": [1, 6], "configur": [1, 3, 4], "confirm": [0, 3], "consid": 0, "construct": [4, 5], "contain": [2, 6], "control": [3, 4, 5, 6], "convent": 0, "core": 6, "correct": [0, 5], "correctli": [0, 5, 6], "correspond": 2, "cpu": 6, "creat": [1, 2, 3], "critic": 0, "ct": [0, 5], "custom": 5, "dada2": 5, "data": [1, 3, 4], "databas": 0, "dataset": [1, 5], "default": [0, 5], "defin": [2, 5], "degener": 0, "demultiplex": 5, "denois": [4, 5], "depend": 6, "design": [4, 5], "detail": 0, "detect": [0, 5], "differ": 3, "digit": 0, "directori": 2, "divers": 4, "doc": 0, "document": [0, 4], "download": [3, 5], "downstream": 5, "dry": 0, "dure": 0, "dynam": 5, "e": [0, 3], "each": [1, 2], "easi": 5, "edit": 5, "editor": 2, "encount": 0, "end": [1, 5, 6], "ensur": [0, 5], "environ": [1, 4], "equival": 1, "error": 6, "everyth": 6, "exampl": [0, 1, 2, 5], "execut": 0, "expect": 6, "explain": 2, "fa": 1, "fastq": [0, 1, 4, 6], "fastqc": [3, 5], "featur": [0, 4], "fetch": 1, "file": [0, 1, 2, 4, 5, 6], "final": 5, "flexibl": 4, "folder": [0, 2], "follow": [0, 3, 5, 6], "forward": 0, "forward_prim": 5, "from": [4, 5, 6], "full": 5, "function": 6, "g": [0, 3], "gccgcggtaa": [0, 5], "gene": [4, 5], "gener": [0, 1, 6], "gggt": [0, 5], "git": 3, "github": 3, "go": 5, "gtg": [0, 5], "guid": 2, "guidanc": 0, "gz": [0, 1, 6], "ha": 5, "have": 2, "help": [0, 1, 4], "here": [1, 2], "how": 2, "http": [0, 3], "hypervari": [2, 4], "i": [0, 4, 5, 6], "id": 5, "ideal": 4, "illumina": 4, "import": [0, 5], "includ": [2, 3, 4, 5], "increas": 6, "infer": 5, "info": 0, "input": 4, "insid": 2, "instal": [0, 4], "instruct": 1, "intermedi": [1, 5], "io": 0, "issu": 0, "iupac": 0, "jupyt": 4, "larg": 4, "layout": 2, "like": 2, "list": 1, "locat": 2, "make": 0, "manual": 5, "master": [1, 5], "match": 0, "md": 1, "mention": 5, "messag": 3, "meta": 4, "microbi": 4, "miss": 0, "modifi": 5, "multi": 4, "multipl": [2, 3], "multiqc": 3, "must": 0, "nano": 2, "need": [2, 5], "notepad": 2, "nucleotid": 0, "number": 6, "occur": 6, "offici": 0, "onc": 5, "org": 0, "organ": 5, "output": [1, 5, 6], "own": [4, 6], "pair": [1, 5, 6], "paired_end": [1, 2, 5], "path": 0, "perform": 6, "phylogenet": 5, "pipelin": [0, 1, 3, 4, 6], "plugin": 0, "pre": [5, 6], "preliminari": 6, "present": 5, "preview": 0, "primer": [4, 6], "print": 3, "prjna": [0, 2], "prjna1232715_1": 2, "prjna123456": 0, "prjna701402_1": 2, "proce": 5, "process": [1, 4, 5], "profil": 4, "project": 5, "provid": [0, 5, 6], "python": 3, "qc": 3, "qiim": [1, 4, 5], "qiime2": [0, 3], "qualiti": [3, 4, 5, 6], "qzv": 5, "raw": [4, 5], "rawread": [1, 2], "re": 2, "read": [1, 5], "readi": 6, "readm": 1, "readthedoc": 0, "recommend": 0, "refer": 0, "region": [0, 2, 4], "regist": 5, "relev": 5, "repositori": [1, 6], "reproduc": 4, "requir": 3, "resid": 1, "resourc": 0, "revers": 0, "reverse_prim": 5, "root": 6, "rrna": [3, 4, 5], "rule": [1, 5], "run": [0, 1, 6], "same": 2, "sampl": [0, 1, 5, 6], "scale": 4, "script": 1, "section": [0, 5], "sequenc": [0, 1, 2, 3, 4], "setup": [3, 6], "sh": [1, 2, 3, 5, 6], "silva": [0, 1, 5], "singl": [1, 5], "single_end": [1, 2, 5], "smooth": 0, "snakefil": [0, 1], "snakemak": [1, 3, 4, 5], "snakemake_env": [3, 6], "specifi": 2, "squar": 0, "sra": [1, 3], "srr": [0, 2, 5], "srr12345678_1": 0, "srr13686819": 2, "srr13686819_1": [1, 6], "srr13686819_2": [1, 6], "srr32595386": 2, "srrid1": 2, "srrid2": 2, "stage": 3, "start": [0, 2], "step": 0, "store": 1, "structur": 5, "subdirectori": 2, "subsequ": 5, "success": [3, 6], "successfulli": 5, "suit": 5, "summari": 5, "support": [2, 4], "sure": 0, "tabl": 4, "tac": [0, 5], "take": 4, "taxonom": [4, 5], "tctaatcc": [0, 5], "technic": 0, "test": 1, "test_data": [1, 5], "text": 2, "thi": [0, 2, 4, 5, 6], "through": 4, "tool": 6, "train": [0, 1, 5], "tree": 5, "trim": [1, 4, 5, 6], "trimmomat": 3, "under": 5, "understand": 1, "us": [0, 2, 3, 4, 5, 6], "usag": 1, "usual": 0, "v4": [0, 5], "v7": 0, "valid": 0, "variabl": 0, "verifi": 6, "version": 0, "via": 5, "vim": 2, "visual": 5, "webpag": 4, "websit": 4, "welcom": 4, "well": 5, "were": 0, "where": 1, "whether": 5, "which": 5, "without": [0, 5], "work": [2, 6], "workflow": [0, 3, 4, 5], "written": [0, 6], "x": [2, 3, 5], "you": [0, 1, 2, 4, 5, 6], "your": [4, 5, 6]}, "titles": ["Additional Information", "Directory Structure Overview", "Fetching Your Own Data", "Installation", "16SAmpliconKit", "Running the Pipeline", "Testing the installation on a example dataset"], "titleterms": {"1": [0, 2, 3, 5, 6], "16sampliconkit": 4, "2": [0, 2, 3, 5, 6], "3": [0, 2, 3, 5], "4": [0, 2, 5], "5": [0, 5], "activ": 6, "add": 2, "addit": 0, "amplicon": 5, "auto": 5, "base": [0, 5], "bioproject": [2, 5], "choos": 2, "classifi": 0, "clone": 3, "conda": 3, "data": [2, 5], "dataset": 6, "directori": [1, 5, 6], "download": 2, "each": 5, "edit": 2, "end": 2, "environ": [3, 6], "error": 0, "exampl": 6, "execut": [2, 3, 5, 6], "express": 0, "fetch": 2, "folder": 5, "format": 0, "gener": 5, "id": 2, "inform": 0, "instal": [3, 6], "make": [2, 3, 5], "name": 0, "navig": [5, 6], "overview": 1, "own": 2, "pair": 2, "pipelin": 5, "primer": [0, 5], "qiim": 0, "raw": 2, "rawread": 5, "read": 2, "region": 5, "regular": 0, "repositori": 3, "requir": 0, "run": [2, 5], "sampl": 2, "script": [2, 3, 5], "sequenc": 5, "set": 3, "singl": 2, "snakefil": [5, 6], "snakemak": [0, 6], "step": [2, 3, 5, 6], "structur": 1, "taxonom": 0, "test": 6, "test_data": 6, "tool": 3, "type": 5, "up": 3, "updat": 5, "your": 2}})